File: HackerRank/Algorithms/Easy/E0056.cpp
/*
Problem Statement: https://www.hackerrank.com/challenges/time-conversion/problem
*/

#include <iostream>
#include <string>
#include <stdio.h>

using namespace std;

void timeConversion(string s) {
    int hr, min, sec;
    bool meridiem = (s.substr(s.length()-2) == "PM") ? true: false;
    sscanf (s.c_str(), "%d:%d:%d", &hr, &min, &sec);
    if(meridiem && hr != 12)
        hr += 12;
    else if(!meridiem && hr == 12)
        hr = 0;
    printf("%02d:%02d:%02d", hr, min, sec);
}

int main()
{
    string s;
    getline(cin,s);
    timeConversion(s);
    return 0;
}

File: HackerRank/Python/Easy/E0056.py
# Problem Statement: https://www.hackerrank.com/challenges/map-and-lambda-expression/problem

cube = lambda x: x**3

def fibonacci(n):
	fib = [0, 1]
	for i in range(2, n):
		fib.append(fib[i - 1] + fib[i - 2])
	return fib[0:n]